riyaz/db.py

```diff
@@ -289,6 +308,50 @@ def set(cls, key, value):
         return kv.save()
 
 
+class Asset(Document):
+    _TABLE = "asset"
+
+    collection: str
+    collection_id: int
+    filename: str
+
+    filesize: Optional[int]
+    created: Optional[datetime]
+    last_modified: Optional[datetime]
+
+    def _get_full_identifier(self):
+        return f"{self.collection}/{self.collection_id}/{self.filename}"
+
+    def _construct_asset_path(self):
+        assert "/" not in self.filename, "filename should not be a path"
+
+        full_identifier = self._get_full_identifier()
+        return Path(config.assets_path) / full_identifier
+
+    def get_url(self):
+        assets_root = "/assets/"
+
+        full_identifier = self._get_full_identifier()
+        return f"{assets_root}{full_identifier}"
+
+    def save_file(self, on_disk_path: Path):
+        assert on_disk_path.is_file()
+
+        asset_path = self._construct_asset_path()
+
+        stat = on_disk_path.stat()
+        filesize, created, last_modified = (
+            stat.st_size, int(stat.st_ctime), int(stat.st_mtime))
+
+        asset_path.parent.mkdir(parents=True, exist_ok=True)
+        shutil.copyfile(on_disk_path, asset_path)
+
+        self.update(
+            filesize=filesize, created=created, last_modified=last_modified)
```

Comment:
I don't think `self.update` saves the changes to the db. Could you please check?

Reply:
Yeah, I was aware of it. This method only copies the data to disk but doesn't save changes to db. The save is being called from outside.

But both should happen together, I'll make that change.

Reply:
`self.update` is not being used here anymore.